content/cloud/variables.md

```diff
@@ -0,0 +1,208 @@
+expected_password = { required = true }
+password = { required = true }
+
+[[component]]
+id = "pw-checker"
+source = "app.wasm"
+[component.trigger]
+route = "/..."
+[component.build]
+command = "spin py2wasm app -o app.wasm"
+[component.config]
+password = "{{ password }}"
+expected_password = "{{ expected_password }}"
+```
+
+## Using Variables in a Spin Application
+
+Now that we have defined our variables and exposed them to our component, lets implement our Spin application. The application should retrieve each variable and validate that the expected password matched the actual password. We will use the Spin `config_config` SDK to get the values. 
+
+<!-- @selectiveCpy -->
+```py
+from spin_http import Response
+from spin_config import config_get
+
+
+def handle_request(request):
+    expected = config_get("expected_password")
+    actual = config_get("password")
+    response = "Access denied"
+    if expected == actual:
+        response = "Access permitted"
+
+    return Response(200,
+                    {"content-type": "text/plain"},
+                    bytes(response, "utf-8"))
+
+```
+
+Lets build and run our application locally to test it out. We will use the [environment variable provider](../spin/dynamic-configuration.md#environment-variable-provider) to set the variable values locally.
+
+<!-- @selectiveCpy -->
+```console
+$ spin build
+$ SPIN_CONFIG_EXPECTED_PASSWORD="123" SPIN_CONFIG_
+PASSWORD="123" spin up
+Logging component stdio to ".spin/logs/"
+
+Serving http://127.0.0.1:3000
+Available Routes:
+  pw-checker: http://127.0.0.1:3000 (wildcard)
+```
+
+Let's query our application to see if we are granted access.
+
+<!-- @selectiveCpy -->
+```console
+$ curl http://127.0.0.1:3000
+Access permitted%
+```
+
+## Deploying the Application to Fermyon Cloud
+
+Let's deploy our application to the cloud with initial values for our variables. All values are stored encrypted.
+
+<!-- @selectiveCpy -->
+```console
+$ spin deploy --variable expected_password="123" --variable password="123"
+Uploading pw_checker version 0.1.0+r7123456...
+Deploying...
+Waiting for application to become ready........... ready
+Available Routes:
+  pw-checker: https://pw-checker-abcdefg.fermyon.app (wildcard)
+```
+
+And once again check if we have access:
+
+<!-- @selectiveCpy -->
+```console
+$ curl https://pw-checker-abcdefg.fermyon.app
+Access permitted%
+```
+
+We're in! Let's now update one of the variable and see how it dynamically changes. Spin has a `cloud` plugin that is used for managing Spin applications in Fermyon cloud. The `variables` subcommand can be used to `set`, `list`, and `delete` application variables:
+
+<!-- @selectiveCpy -->
+```console
+$ spin cloud variables --help
+cloud-variables 0.1.0
+Manage Spin application variables
+
+USAGE:
+    cloud variables <SUBCOMMAND>
+
+OPTIONS:
+    -h, --help       Print help information
+    -V, --version    Print version information
+
+SUBCOMMANDS:
+    delete    Delete variable pairs
```

Comment:
What are "variable pairs" in this context?

Reply:
the variable name and value. But i guess you're saying in the context of delete we should probably just say "Delete a variable"

Reply:
Gotcha.  Yeah sorry I dropped the ball on that review.  It's not a problem and we can iterate on the text separately.